[2025-06-18T11:23:58.526+0900] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-06-18T11:23:58.550+0900] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: image_preprocess_v1.img_preprocess_task manual__2025-06-18T02:23:52.961100+00:00 map_index=1 [queued]>
[2025-06-18T11:23:58.563+0900] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: image_preprocess_v1.img_preprocess_task manual__2025-06-18T02:23:52.961100+00:00 map_index=1 [queued]>
[2025-06-18T11:23:58.564+0900] {taskinstance.py:2867} INFO - Starting attempt 1 of 1
[2025-06-18T11:23:58.578+0900] {taskinstance.py:2890} INFO - Executing <Mapped(_PythonDecoratedOperator): img_preprocess_task> on 2025-06-18 02:23:52.961100+00:00
[2025-06-18T11:23:58.587+0900] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=23790) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-06-18T11:23:58.589+0900] {standard_task_runner.py:72} INFO - Started process 23834 to run task
[2025-06-18T11:23:58.593+0900] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'image_preprocess_v1', 'img_preprocess_task', 'manual__2025-06-18T02:23:52.961100+00:00', '--job-id', '974', '--raw', '--subdir', 'DAGS_FOLDER/test_dag.py', '--cfg-path', '/tmp/tmpbrm7b58x', '--map-index', '1']
[2025-06-18T11:23:58.597+0900] {standard_task_runner.py:105} INFO - Job 974: Subtask img_preprocess_task
[2025-06-18T11:23:58.647+0900] {task_command.py:467} INFO - Running <TaskInstance: image_preprocess_v1.img_preprocess_task manual__2025-06-18T02:23:52.961100+00:00 map_index=1 [running]> on host f6d4228a9eb8
[2025-06-18T11:23:58.734+0900] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='image_preprocess_v1' AIRFLOW_CTX_TASK_ID='img_preprocess_task' AIRFLOW_CTX_EXECUTION_DATE='2025-06-18T02:23:52.961100+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-18T02:23:52.961100+00:00'
[2025-06-18T11:23:58.736+0900] {taskinstance.py:732} INFO - ::endgroup::
[2025-06-18T11:23:58.741+0900] {logging_mixin.py:190} INFO - empty map check {} {'process_id': '_bc6f2817-2122-4a90-8f7f-dc69abb2aa40_pre', 'folder_path': '931319ff-8d1f-46aa-9817-c6d39268562b/bc6f2817-2122-4a90-8f7f-dc69abb2aa40', 'step_list': [{'name': 'calc_angle_set2', 'param': {'key': 'angle1', 'iterations': 4, 'iter_save': True}}, {'name': 'text_orientation_set', 'param': {'key': 'angle2', 'iterations': 4, 'iter_save': True}}, {'name': 'line_tracking', 'param': {'iter_save': True}}, {'name': 'save', 'param': {'key': 'classify_preprocess'}}], 'result_file_list': []}
[2025-06-18T11:23:58.742+0900] {logging_mixin.py:190} INFO - step : calc_angle_set2
[2025-06-18T11:24:07.812+0900] {logging_mixin.py:190} INFO - osd 1 0.0 0.0 20 74748 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 223, 12320, 42664, 74748, 43644, 10201, 1061, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
[2025-06-18T11:24:16.794+0900] {logging_mixin.py:190} INFO - osd 2 0.0 0.0 20 74748 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 223, 12320, 42664, 74748, 43644, 10201, 1061, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
[2025-06-18T11:24:26.134+0900] {logging_mixin.py:190} INFO - osd 3 0.0 0.0 20 74748 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 223, 12320, 42664, 74748, 43644, 10201, 1061, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
[2025-06-18T11:24:35.836+0900] {logging_mixin.py:190} INFO - osd 4 0.0 0.0 20 74748 [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 223, 12320, 42664, 74748, 43644, 10201, 1061, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
[2025-06-18T11:24:35.880+0900] {logging_mixin.py:190} INFO - step : text_orientation_set
[2025-06-18T11:24:41.368+0900] {logging_mixin.py:190} INFO - file_copy: /opt/***/data/temp/931319ff-8d1f-46aa-9817-c6d39268562b/bc6f2817-2122-4a90-8f7f-dc69abb2aa40/rotate2_1.png
[2025-06-18T11:24:43.304+0900] {logging_mixin.py:190} INFO - osd 1 270 270 5.35 Latin 1.73
[2025-06-18T11:24:47.591+0900] {logging_mixin.py:190} INFO - file_copy: /opt/***/data/temp/931319ff-8d1f-46aa-9817-c6d39268562b/bc6f2817-2122-4a90-8f7f-dc69abb2aa40/rotate2_2.png
[2025-06-18T11:24:50.978+0900] {logging_mixin.py:190} INFO - osd 2 450 180 3.32 Cyrillic 2.35
[2025-06-18T11:24:55.587+0900] {logging_mixin.py:190} INFO - file_copy: /opt/***/data/temp/931319ff-8d1f-46aa-9817-c6d39268562b/bc6f2817-2122-4a90-8f7f-dc69abb2aa40/rotate2_3.png
[2025-06-18T11:24:58.319+0900] {logging_mixin.py:190} INFO - osd 3 break  450 0 2.85 Latin 4.24
[2025-06-18T11:24:58.320+0900] {logging_mixin.py:190} INFO - step : line_tracking
[2025-06-18T11:24:58.325+0900] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/decorators/base.py", line 266, in execute
    return_value = super().execute(context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/tasks/img_preprocess_task.py", line 64, in img_preprocess_task
    output = function_info["function"](input,**stepinfo["param"])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/tasks/img_preprocess_task.py", line 633, in line_tracking
    horizontal = binary.copy()
                 ^^^^^^^^^^^
AttributeError: 'tuple' object has no attribute 'copy'
[2025-06-18T11:24:58.337+0900] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=image_preprocess_v1, task_id=img_preprocess_task, run_id=manual__2025-06-18T02:23:52.961100+00:00, map_index=1, execution_date=20250618T022352, start_date=20250618T022358, end_date=20250618T022458
[2025-06-18T11:24:58.361+0900] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-06-18T11:24:58.362+0900] {standard_task_runner.py:124} ERROR - Failed to execute job 974 for task img_preprocess_task ('tuple' object has no attribute 'copy'; 23834)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/decorators/base.py", line 266, in execute
    return_value = super().execute(context)
                   ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/tasks/img_preprocess_task.py", line 64, in img_preprocess_task
    output = function_info["function"](input,**stepinfo["param"])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/tasks/img_preprocess_task.py", line 633, in line_tracking
    horizontal = binary.copy()
                 ^^^^^^^^^^^
AttributeError: 'tuple' object has no attribute 'copy'
[2025-06-18T11:24:58.386+0900] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-06-18T11:24:58.409+0900] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-06-18T11:24:58.414+0900] {local_task_job_runner.py:245} INFO - ::endgroup::
